### DISCUSSION TITLE
inventory add products in btches

### DISCUSSION THREAD

**Discussion by rioshaz** [YetiForceCompany/YetiForceCRM#14431] [General]
Hi guys does anyone know how to add same product in different batches,,
suppose :
A medicine is added as a products "paracetamol" and now this medicine comes again and again in different batches then how to add it in to system ? 
 
Every batch has their own price and quantity.

ideally we should be able to see how much batches are available for a medicine with their price n quantity

**Comment by vovpff** [comment]
I think you need separate products of this drug like 10 pc paracetamol, 24 pc paracetamol and etc

**Comment by rioshaz** [comment]
yes thats ok.. but 
suppose i add 10 pc parateamol with batch no.. 1234 

then i again get 10 pc more parateamol with batch no. 5678 

how to i enter that... a product can have only 1 batch no...

problem is everytime i have to add all details again n again for paracetamol like name, manufacutere etc... this is very unreal,,,

after this how do i  update product quantity:

good recieved note module manage products quanity, so if i want to update paracetamol quantity product it will not udpate batch no.. 
i want to use stock quanity as per medicine and then how many batch 1 medine have,,,,

thats the issue...

you know any solution ?

**Reply by vovpff** [reply]
Ok. I understand what you want. I think there is no way to achieve this easy. This can be implemented by programmer or Yetiforce team.

**Comment by rioshaz** [comment]
@vovpff but its so basic functionality every company has inventory in batches and yf being a powerful system lacks such a basic option i am astonished... truely disappointing ...even basic crm have this,,

**Reply by vovpff** [reply]
I don't know CRM systems what has this functionality from the box

**Comment by rioshaz** [comment]
erpnext

**Reply by mariuszkrzaczkowski** [reply]
then why don't you use this ERP?  you're just here complaining

**Reply by rioshaz** [reply]
because 

> then why don't you use this ERP? you're just here complaining

because  its very complex and yf is also same complex but i have had enough time so know basic stuff...


also there is no point in feeling bad about it.. rather u should try to learn n offer a solution to your users... 
products now have batch no. functionality kills the whole inventory system...

i cannot always add same product again n again....

pls help if u can

**Reply by rioshaz** [reply]
i have also contributed to many hours  to yf n given u quite a few error n have been ur free tester,, lets support each other not  misunderstand

**Comment by vovpff** [comment]
Ok. I'm noticed this!

**Reply by rioshaz** [reply]
is there any way you think we can have a solution how to add batches to products...  anything u would do if in my place..

**Reply by mariuszkrzaczkowski** [reply]
you have two 2 options: write it yourself or pay someone

**Reply by rioshaz** [reply]
yes pay ,, pls tell me cost...

**Reply by mariuszkrzaczkowski** [reply]
nice functionality, but few systems have it.  give me a bigger description, we will quote

**Reply by rioshaz** [reply]
> nice functionality, but few systems have it. give me a bigger description, we will quote

@mariuszkrzaczkowski yes sure i can try to explain..
suppose i add a product .. example a medicine called "drug"
now first time i enter it.. i can give name,batch no. , expry date, manufacturer  etc then i go to good recieved note module and enter quantity recieved ex: 10 ,, thats fine now the product called drug is updated with all details and quanity in stock is 10 .

problem is when next time i received same medicine,, but now its just a different batch no, with new costs, purchase price, expiry date and quantity, how to i add i ?

there is no way to add batches in a product.. 

so if now i make a sales invoice for the product :drug: i should be able to see quantity in each batch and their expry date .. so i can pick the right batch n sell the one expirying earliest.. 

basically how a medical shop sells medicines,, based on batch no, n then it gets deducted from stock too.
